### PR TITLE
feat(project information): add pre exam reps logic

### PR DESCRIPTION
**Pull request by jonathangoulding**

### CODE REVIEW

File: packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	if (dayToday.isAfter(dayOpenDate) && (dayToday.isBefore(dayClosedDate) || closedDate === null))
+		subStages.OPEN_REPS = true;
+
+	if (dayToday.isAfter(dayClosedDate) && dayToday.isBefore(dayWebsiteDate))
+		subStages.CLOSED_REPS = true;
+
+	if (rule6) {
+		if (dayToday.isAfter(dayWebsiteDate) && rule6) subStages.RULE_6_PUBLISHED_REPS = true;
+	} else {
+		if (dayToday.isAfter(dayWebsiteDate)) subStages.PUBLISHED_REPS = true;
```
Comment (jonathangoulding) [inline packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js:29]: "PRE_REPS":false,"OPEN_REPS":true,"CLOSED_REPS":false,"PUBLISHED_REPS":true,"RULE_6_PUBLISHED_REPS":false

Possible with dates -  Open (2017-02-14) closed (2024-08-01) websote (2021-08-17) false

Rep open and published both true

Comment (jonathangoulding) [reply packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js:29]: test is fine. But when rendered it shows true.

**Review by Yakitrak — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments

File: packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		PRE_REPS: false,
+		OPEN_REPS: false,
+		CLOSED_REPS: false,
+		PUBLISHED_REPS: false,
+		RULE_6_PUBLISHED_REPS: false
+	};
+
+	if (dayToday.isBefore(dayOpenDate) || openDate === null) subStages.PRE_REPS = true;
+
+	if (dayToday.isAfter(dayOpenDate) && (dayToday.isBefore(dayClosedDate) || closedDate === null))
```
[the remark this answers is not in the document]
Comment (Yakitrak) [reply packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js:19]: Do we have logic somewhere to catch this empty page?

[the remark this answers is not in the document]
Comment (jonathangoulding) [reply packages/forms-web-app/src/pages/projects/project-information/utils/pre-examination-sub-stages.js:19]: The description just wouldn't show. 

`<p class="govuk-body">
	{% if preExamSubStages.PRE_REPS %}
		{{ preRepsDescText | safe }}
	{% elseif preExamSubStages.OPEN_REPS %}
		{{ repsOpenDescText | safe }}
	{% elseif preExamSubStages.CLOSED_REPS %}
		{{ repsClosedDescText | safe }}
	{% elseif preExamSubStages.PUBLISHED_REPS %}
		{{ repsPublishedDescText | safe }}
	{% elseif preExamSubStages.RULE_6_PUBLISHED_REPS %}
		{{ preLimDescText | safe }}
	{% endif %}
</p>
`

File: packages/forms-web-app/src/pages/services.js [hunk 3]
```diff
@@ -8,6 +8,12 @@ const getProjectUpdatesData = async (caseRef) => {
 	return response.data.updates;
 };
 
+const getRule6DocumentType = async (case_ref) => {
+	const { data, resp_code } = await getDocumentByType(case_ref, 'RULE_6_LETTER');
```
[the remark this answers is not in the document]
Comment (jonathangoulding) [reply packages/forms-web-app/src/pages/services.js:12]: Can lift it to a const in this file for now then move out at a later time when needed.

**Review by Yakitrak — APPROVED** [review]

**Pull request merged**